_tests_/pet.test.js

```diff
@@ -0,0 +1,155 @@
+  it('increases fitness by 4 to a maximum of 10', () => {
+    const pet = new Pet('fido');
+
+    pet.fitness = 8;
+    pet.walk();
+
+    expect(pet.fitness).toEqual(10);
+  });
+  it("walk does not function if pet is not alive" ,() => {
+    const pet = new Pet("fido");
+    pet.fitness = -1;
+    expect(() => pet.walk()).toThrow("your pet is no longer alive :(");
+  });
+});
+
+describe('feed', () => {
+  it('decreases the pets hunger levels', () => {
+    const pet = new Pet('fido');
+
+    pet.hunger = 0;
+    pet.feed();
+
+    expect(pet.hunger).toEqual(0);
+  });
+  it("feed does not function if pet is not alive" ,() => {
+    const pet = new Pet("Fido");
+    pet.hunger = 30;
+    expect(() => pet.feed()).toThrow("your pet is no longer alive :(");
+  });
+});
+
+describe('checkUp', () => {
+  it('check if pet needs a walk', () => {
+    const pet = new Pet('fido');
+
+    pet.fitness = 3
+    expect(pet.checkUp()).toBe('I need a walk');
+  });
+  it('check if pet is hungry', () => {
+    const pet = new Pet('fido');
+    pet.hunger = 5;
+    expect(pet.checkUp()).toEqual('I am hungry');
+   });
+  it('check if pet needs both walking and feeding', () => {
+    const pet = new Pet('fido');
+    pet.fitness = 3;
+    pet.hunger = 6;
+    expect(pet.checkUp()).toEqual('I am hungry AND I need a walk');
+   });
+   it('if pet is neither hungry nor in need of a walk', () => {
+    const pet = new Pet('fido');
+   pet.fitness = 6;
+   pet.hunger = 3;
+   expect(pet.checkUp()).toEqual('I feel great'); 
+  });
+  it("checkup does not function if pet is not alive" ,() => {
+    const pet = new Pet("Fido");
+    pet.age = 30;
+    expect(() => pet.growUp()).toThrow("your pet is no longer alive :(");
+  });
+});
+
+describe('death', () => {
+  it('should be alive when growing up and walking', () => {
+    const pet = new Pet('Fido');
+    pet.growUp();
+    expect(pet.isAlive).toEqual(true);
+
+    pet.walk();
+    expect(pet.isAlive).toEqual(true);
+  });
+  it('check if pets fitness is 0', () => {
+    const pet = new Pet('fido');
+    pet.fitness = 0;
+    expect(pet.isAlive).toEqual(false);
+  });
+  it('check if pets hunger is 10 or more', () => {
+    const pet = new Pet('fido');
+    pet.hunger = 10;
+    expect(pet.isAlive).toEqual(false);
+  });
+  it('check if pets age is 30', () => {
+    const pet = new Pet('fido');
+    pet.age = 30;
+    expect(pet.isAlive).toEqual(false);
+  });
+
+describe('Pet Adoption', () => {
+  it('Parent pet adopts a child', () => {
+    const parentPet = new Pet('Dave');
+    const childPet = new Pet('Amelia')
+    
+    parentPet.adoptChild(childPet);
+    
+    expect(parentPet.children[0]).toEqual(childPet);
+  });
+
+  it("haveBaby", () => {
+    const parent = new Pet("Thom");
+    
+    expect(parent.children).toEqual([]);
+  });
+
+  it("haveBaby", () => {
+    const parent = new Pet("Thom");
+    parent.haveBaby("Annie");
+
+    expect(parent.children[0].name).toEqual("Annie");
+  });
```

Comment:
I would also add an `expect` statement to check that `parent.chilren[0]` is a instance of the Pet class. Otherwise, this test could pass if your child was just an object with a matching name property. e.g.
```
it("haveBaby", () => {
  const parent = new Pet("Thom");
  parent.haveBaby("Annie");

  expect(parent.children[0]).toBeInstanceOf(Pet);
  expect(parent.children[0].name).toEqual("Annie");
});
```